packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx

```diff
@@ -104,7 +104,7 @@ const GatsbyImageHydrator: FC<GatsbyImageProps> = function GatsbyImageHydrator({
 
     // The plugin image component is a bit special where if it's server-side rendered, we add extra script tags to support lazy-loading without
     // In this case we stop hydration but fire the correct events.
-    const ssrImage = root.current.querySelector(
+    const ssrImage = root.current?.querySelector(
```

Comment:
You don't need the question mark here as it's in the `useEffect` and it'll always be available